packages/grafana-ui/src/components/Forms/Field.tsx

```diff
@@ -22,7 +22,7 @@ export interface FieldProps extends HTMLAttributes<HTMLDivElement> {
   /** Indicates if field is required */
   required?: boolean;
   /** Error message to display */
-  error?: string | null;
+  error?: React.ReactNode;
```

Comment:
Hey @joshhunt 👋🏾 , I am lacking to understand how this change is related to the translations 😅 , what kind of react node should we get?  If you don't mind can you explain the flow 🥺

Reply:
Great question! See https://github.com/grafana/grafana/pull/43874/files#diff-33f818cdcc6583f4bc4c19f60662c43076f04688cc71187e46f379378b2dc09bR30

Separate from i18n work, whenever you have a react prop that is displayed in the DOM like `{error}`...

https://github.com/grafana/grafana/blob/1cab71456cb8ba7fd982f0b340e53073f6ff27b6/packages/grafana-ui/src/components/Forms/Field.tsx#L108

...I think it is good practice to type that as `React.ReactNode` rather than strictly a string so our components are more composable. It allows consumers to pass in whatever react-renderable content they need to basically for free.

For I18n, the _main_ way we translate phrases is with the Trans component - `<Trans id="user-profile.fields.name-error">Name is required</Trans>`. We need props to accept `React.ReactNode` in order to pass this in.

Reply:
Thank you! Now I get it 🙌🏾